packages/rum-core/src/domain/rumEventsCollection/view/trackViewMetrics.spec.ts

```diff
@@ -138,6 +138,44 @@ describe('rum track view metrics', () => {
       expect(getViewUpdateCount()).toEqual(2)
       expect(getViewUpdate(1).loadingTime).toEqual(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
     })
+
+    it('should use computed loading time from time origin for initial view', () => {
+      // introduce a gap between time origin and tracking start
+      // ensure that `load event > activity delay` and `load event < activity delay + clock gap`
+      // to make the test fail if the clock gap is not correctly taken into account
+      const CLOCK_GAP =
+        FAKE_NAVIGATION_ENTRY_WITH_LOADEVENT_AFTER_ACTIVITY_TIMING.loadEventEnd -
+        BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY +
+        1
+
+      setupBuilder.beforeBuild((buildContext) => {
+        // need to restart the view test to introducing a clock gap before the tracking start
+        viewTest.stop()
+        buildContext.clock.tick(CLOCK_GAP)
+        viewTest = setupViewTest(buildContext)
+        return viewTest
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏This feels a bit confusing. What about either:
* always advance the clock before calling `setupViewTest`, as it could reflect a more realistic situation (the SDK will never be initialized at "navigation start"/relative time 0)
* exposing the clock as a `TestSetupBuilder` property, so we can do:
```ts
setupBuilder.clock.tick(CLOCK_GAP)
const { ... } = setupBuilder.build()
```
* using the `mockClock` function directly instead of using `setupBuilder.withFakeClock()`

Reply:
I went with the second one in [f499364](https://github.com/DataDog/browser-sdk/pull/1315/commits/f49936432d4e81fffc273035661014653be4ebac)